### PR TITLE
New melting parameterisation

### PR DESCRIPTION
**Pull request by boriskaus**
This implements a smooth melting parameterisation used by Oleg Melnik and colleagues, appropriate for basalt (blue) and rhyolite (red curve)
MeltingParam_Melnik

### CODE REVIEW

File: src/MeltFraction/MeltingParameterization.jl [hunk 1]
```diff
@@ -89,6 +90,80 @@ function show(io::IO, g::MeltingParam_Caricchi)
...
+    return MaterialParamsInfo(;
+        Equation=L"\phi = f(T)"
+    )
+end
+
+# Calculation routines
+function (p::MeltingParam_Melnik)(; T, kwargs...)
+    @unpack_val a, b, c, d, Tchar = p
+    x = T / Tchar
+    θ = a + b * x + c * x^2 + d * x^3;
```
Comment (albert-de-montserrat) [inline src/MeltFraction/MeltingParameterization.jl:146]:
```suggestion
    θ = a + b * x + c * x^2 + d * x^3;
```

Could you use `evalpoly` here? it's much more efficient to evaluate polynomials as it uses Horner's rule. I.e.

```julia
θ = evalpoly(x, a, b , c , d)
```

File: src/MeltFraction/MeltingParameterization.jl [hunk 2]
```diff
@@ -89,6 +90,80 @@ function show(io::IO, g::MeltingParam_Caricchi)
...
+    ϕ = inv(1.0 + exp(θ))
+    return ϕ
+end
+
+function compute_dϕdT(p::MeltingParam_Melnik; T, kwargs...)
+    @unpack_val a, b, c, d, Tchar = p
+    
+    x = T / Tchar
+
+    dϕdT = -((2T*c) / (Tchar^2) + b / Tchar + (3d*((x)^2)) / Tchar)*(1 / ((1.0 + exp(a + (T*b) / Tchar + c*((x)^2) + d*((x)^3)))^2))*exp(a + (T*b) / Tchar + c*((x)^2) + d*((x)^3))
```
Comment (albert-de-montserrat) [inline src/MeltFraction/MeltingParameterization.jl:156]:
```suggestion
    dϕdT = -((2T*c) / (Tchar^2) + b / Tchar + (3d*((x)^2)) / Tchar)*(1 / ((1.0 + exp(a + (T*b) / Tchar + c*((x)^2) + d*((x)^3)))^2))*exp(a + (T*b) / Tchar + c*((x)^2) + d*((x)^3))
```
May be worth to pre-compute some repeated values here; may help also with readability. And many brackets can be removed.

**Review by albert-de-montserrat — CHANGES_REQUESTED** [review]

File: src/MeltFraction/MeltingParameterization.jl (hunk 1 above)
Comment (boriskaus) [reply src/MeltFraction/MeltingParameterization.jl:146]: nice, didn' know that

File: src/MeltFraction/MeltingParameterization.jl (hunk 2 above)
Comment (boriskaus) [reply src/MeltFraction/MeltingParameterization.jl:156]: good point

**Review by albert-de-montserrat — APPROVED** [review]

**Comment by albert-de-montserrat** [comment]
Looks good to me !

**Pull request merged**